QMLComponents/analysisform.cpp

```diff
@@ -272,16 +273,28 @@ void AnalysisForm::setHasVolatileNotes(bool hasVolatileNotes)
 	emit hasVolatileNotesChanged();
 }
 
-bool AnalysisForm::parseOptions(Json::Value &options)
+
+QString AnalysisForm::parseOptions(QString options)
 {
-	if (_rSyntax->parseRSyntaxOptions(options))
+	Json::Reader jsonReader;
+	Json::Value	 jsonOptions;
+	Json::Value jsonResult(Json::objectValue);
+
+	QJsonDocument doc = QJsonDocument::fromJson(options.toUtf8());
+	jsonReader.parse(doc.toJson().toStdString(), jsonOptions, false);
+
+	if (!_analysis)
+		setAnalysis(new AnalysisBase(this)); // Create a dummy analyis object
+
+	if (_rSyntax->parseRSyntaxOptions(jsonOptions))
 	{
-		bindTo(options);
-		options = _analysis->boundValues();
-		return true;
+		bindTo(jsonOptions);
+		jsonOptions = _analysis->boundValues();
 	}
 
-	return false;
+	jsonResult["options"] = jsonOptions;
+	jsonResult["error"] = fq(getError());
+	return tq(jsonResult.toStyledString());
```

Comment:
Why not return the Json value then?

Reply:
In the jaspQMLR example, the QML form is just a QuickItem object, so I call the parseOptions with the QMetaObject::invokeMethod method:
https://github.com/jasp-stats/jaspQmlR/blob/a7a93a2f6fa103d47560c1ed541e9c252eed3671/src/qmlR.cpp#L208
In this way I don't need to load the AnalysisFormBase header in the jaspQMLR package.
I don't think I can pass a Json::Value argument in the InvokeMethod, so I thought just send the string, and reset it as Json object afterwards.